perf-tool/src/monitor.cpp

```diff
@@ -48,96 +49,66 @@ JNIEXPORT void JNICALL MonitorContendedEntered(jvmtiEnv *jvmtiEnv, JNIEnv *env,
+        {
+            env->ExceptionDescribe();
+            env->ExceptionClear();
+        }
+        /* Now get the class object's class descriptor */
+        cls = env->GetObjectClass(clsObj);
+        /* Find the getName() method on the class object */
+        mid = env->GetMethodID(cls, "getName", "()Ljava/lang/String;");
+        if (mid)
         {
-            auto jMethods = json::array();
-            bool error = false;
-            for (int i=0; i < count && !error; i++)
+            /* Call the getName() to get a jstring object back */
+            jstring strObj = (jstring)env->CallObjectMethod(clsObj, mid);
+            if (env->ExceptionCheck() == JNI_TRUE)
             {
-                char *methodName;
-                char *signature;
-                err = jvmtiEnv->GetMethodName(frames[i].method, &methodName, &signature, NULL);
-                if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve Method Name.\n")) 
-                {
-                    jclass declaring_class;
-                    err = jvmtiEnv->GetMethodDeclaringClass(frames[i].method, &declaring_class);
-                    if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve Method Declaring Class.\n")) 
-                    {
-                        char *declaringClassName;
-                        err = jvmtiEnv->GetClassSignature(declaring_class, &declaringClassName, NULL);
-                        if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve Method Declaring Class Signature.\n")) 
-                        {
-                            json jMethod;
-                            jMethod["class"] = declaringClassName;
-                            jMethod["method"] = methodName;
-                            jMethod["signature"] = signature;
-                            jMethods.push_back(jMethod);
-                            err = jvmtiEnv->Deallocate((unsigned char*)declaringClassName);
-                            check_jvmti_error(jvmtiEnv, err, "Unable to deallocate class name.\n");
-                        }
-                        else
-                        {
-                            error = true;
-                        }
-                    }
-                    else
-                    {
-                        error = true;
-                    }
-                    err = jvmtiEnv->Deallocate((unsigned char*)methodName);
-                    check_jvmti_error(jvmtiEnv, err, "Unable to deallocate methodName.\n");
-                    err = jvmtiEnv->Deallocate((unsigned char*)signature);
-                    check_jvmti_error(jvmtiEnv, err, "Unable to deallocate method signature.\n");
-                }
-                else
-                {
-                    error = true;
-                }
-            } /* end for loop */
-            j["stackTrace"] = jMethods;
+                env->ExceptionDescribe();
+                env->ExceptionClear();
+            }
```

Comment:
In an exception has occurred, I would treat that as an error case and we should not rely on the output of the function, but rather abort.